src/Jellyfin.Mvvm/Services/StateService.cs

```diff
@@ -7,20 +7,45 @@ public class StateService : IStateService
 {
     private readonly CurrentStateModel _state;
     private readonly SdkClientSettings _sdkClientSettings;
+    private readonly IStateStorageService _stateStorageService;
 
     /// <summary>
     /// Initializes a new instance of the <see cref="StateService"/> class.
     /// </summary>
     /// <param name="sdkClientSettings">Instance of the <see cref="SdkClientSettings"/>.</param>
-    public StateService(SdkClientSettings sdkClientSettings)
+    /// <param name="stateStorageService">Instance of the <see cref="IStateStorageService"/> interface.</param>
+    public StateService(SdkClientSettings sdkClientSettings, IStateStorageService stateStorageService)
     {
         _sdkClientSettings = sdkClientSettings;
+        _stateStorageService = stateStorageService;
         // TODO load from disk
         _state = new CurrentStateModel();
         _sdkClientSettings.BaseUrl = _state.Host;
         _sdkClientSettings.AccessToken = _state.Token;
     }
 
+    /// <inheritdoc/>
+    public async ValueTask InitializeAsync()
+    {
+        var storeState = await _stateStorageService.GetStoredStateAsync().ConfigureAwait(false);
+
+        var selectedServer = storeState.Servers.FirstOrDefault(x => x.Id == storeState.SelectedServerId);
```

Comment:
Should only populated `selectedServer` if `storeState.SelectedServerId` isn't null.
Same for `selectedUser`